### ISSUE TITLE
DIV/IDIV should trap on all overflow, not just divide by zero

### ISSUE THREAD

**Issue by neldredge** [YJDoc2/8086-Emulator#13]
If IDIV would produce a quotient of 0x8000 (word) or 0x80 (byte), the real 8086 traps (INT 0, divide overflow).  8086-Emulator yields 0x8000 and continues.  (Note this behavior changed between 8086 and 80386, not sure what the 80186/80286 did.)

**Comment by neldredge** [comment]
In fact the emulator only traps on `div/idiv` when the divisor is actually zero.  It should trap whenever the mathematical result does not fit in the destination register.  For instance
```
mov ax, 0xffff
mov dx, 0xffff
mov bx, 1
div bx
```
should trap with INT 0.

**Comment by YJDoc2** [comment]
Yes, this is correct. When implementing, apparently I read the spec incorrectly, and thought it was specifically for 0
